### DISCUSSION TITLE
Request for a sample that shows how to do object picking with the mouse.

### DISCUSSION THREAD

**Discussion by metric-tensor** [MethanePowered/MethaneKit#97] [Ideas]
I think it would be helpful to have a sample showing mouse picking/selection.

**Comment by zhangfq-chemistry** [comment]
The same request! Thanks!

**Comment by egorodet** [comment]
Folks, this is a graphics rendering engine. Don't confuse rendering engine with game engine. I don't plan to add any kind of entity management system and scene editor for manipulating objects, since this library is not supposed to become game engine.

Anyway simple object picking can be accomplished fully on CPU and does not involve any rendering at all. Just cast a ray passing through the camera eye and a mouse cursor on projection plane transformed to world coordinates, then traverse your scene objects ordered from front to back and find first object intersecting with you mouse ray. Hope this would help.
